src/code/sys_freeze.c

```diff
@@ -4,7 +4,13 @@
 #include "sys_freeze.h"
 #include "terminal.h"
 
-NORETURN void func_800D31A0(void) {
+/**
+ * OOT uses this a few times to halt execution when a critical error/failure occurs that
+ * cannot be recovered from (though all occurences are in N64DD-related codepaths).
+ * Note that this function only freezes the main thread, other threads will continue to run
+ * (e.g. the audio thread).
+ */
```

Comment:
We generally don't write "OoT does/uses x" in code comments. We also typically write comments from the perspective of a "real" codebase, as in comments should describe to a prospective user of this function what it does and how it might be used; notes on how it's "actually" used in practice tend to be more of a footnote.

Finally there's a small distinction to be made: "main thread" refers to **the** main thread, but this function freezes the thread that calls into it. From a quick look, the "graph" thread calls this function in at least the console logo cases.

Something like this:
```suggestion
/**
 * Halts the calling thread forever, without destroying it or affecting other threads.
 * This may be used to halt execution in the event of an irrecoverable error without triggering
 * the crash screen or otherwise halting the rest of the program, other threads continue
 * as normal.
 *
 * @note In practice this approach is rarely used, only occurring in N64DD-related
 *       codepaths. In most cases it is more appropriate to trigger the crash screen
 *       with Fault_AddHungupAndCrash or related functions.
 *
 * @see Fault_AddHungupAndCrash
 */
```

Reply:
I've changed the commit to have the appropriate changes as requested.